2021/CifarelliDiFranco/dla.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python3
+
+import subprocess
+import time
+import threading
+from influxdb import InfluxDBClient
+import utils
+import pyshark
+import os
+import signal
+import shutil
+import datetime
+import numpy
+import traceback
+
+args = utils.create_parser()
+
+IP_resps = {}
+values = {}
+
+def shutdown(snifferpid):
+    while True:
+        user_input = input("type quit (only) when the capture has to arrest\n")
+        if user_input == "quit":
+            os.kill(snifferpid, signal.SIGINT)
+            break
+        else: continue
+
+def main():
+    if not os.path.isdir(utils.PCAP_PATH): os.mkdir(utils.PCAP_PATH)
+    if not os.path.isdir(utils.LIVECAPTURE_PATH): os.mkdir(utils.LIVECAPTURE_PATH)
+    if not os.path.isdir(utils.MERGED_PATH): os.mkdir(utils.MERGED_PATH)
+    if not os.path.isdir(utils.OUTPUT_PATH): os.mkdir(utils.OUTPUT_PATH)
+    if not os.path.isdir(utils.DEBUG_PATH): os.mkdir(utils.DEBUG_PATH)
+
+    if args.postanalysis and not args.report:
+        print("it's not possible to do a post analysis without creating a report (flag -pa set but flag -r not set)")
+        return
+
+    influx_client = InfluxDBClient(username="students", password="gr21", database="progettogr")
+
+    ipv4, subnet = utils.get_net_info(args.interface)
+    print("sniffing on "+args.interface+" (IPv4 = "+ipv4+", subnet = "+subnet+")")
+
+    if not args.ignore:
```

Comment:
Tutte queste scritture di files come root sono molto pericolose

Reply:
Come specificato nel README:
"Before starting the software itself remember to give capabilities to your binaries.
On our testing machine this capability was enough to run the software: sudo setcap cap_net_raw=eip /usr/bin/python3."